plugin/pkg/admission/gc/gc_admission.go

```diff
@@ -99,10 +120,93 @@ func isChangingOwnerReference(newObj, oldObj runtime.Object) bool {
 	return false
 }
 
+// translates ref to a DeleteAttribute deleting the object referred by the ref.
+func (a *gcPermissionsEnforcement) toDeleteAttribute(ref metav1.OwnerReference, attributes admission.Attributes) (authorizer.AttributesRecord, error) {
+	groupVersion, err := schema.ParseGroupVersion(ref.APIVersion)
+	if err != nil {
+		return authorizer.AttributesRecord{}, err
+	}
+	mapping, err := a.restMapper.RESTMapping(schema.GroupKind{Group: groupVersion.Group, Kind: ref.Kind}, groupVersion.Version)
```

Comment:
> I think it's a bug in the design phase that the ownerReference records the Kind instead of Resource. I don't think we should fix it by trying all restmappings, rather, we should add a Resource field to the ownerReference. WDYT?

I have long thought that, but its controversial.  In other places, @liggitt wants to keep kind assumes a canonical mapping from one to the other.  I can't tell what @lavalamp wants given comments in the TPR thread, but I previously thought he liked the one to one mapping.

Given that there is more than one mapping possible, we can start out strict (must have delete access to all) and make it less strict later if we find it to be a problem.

I would not start down the path of changing the API without broad agreement.